src/modules/NtpClient.rb

```diff
@@ -509,48 +509,24 @@ def GetUsedNtpServers
         used_servers << record["address"] if record["type"] == "server"
       end
 
-      deep_copy(used_servers)
+      used_servers
     end
 
     # Checks whether all servers listed in the random_pool_servers list
     # are used in the configuration.
     #
     # @return [Boolean] true if enabled
     def IsRandomServersServiceEnabled
-      # all servers needed by pool.ntp.org service, before checking false == not used
-      needed_servers = {}
-      Builtins.foreach(@random_pool_servers) do |server_name|
-        Ops.set(needed_servers, server_name, false)
-      end
+      used_servers = GetUsedNtpServers()
 
-      Builtins.foreach(GetUsedNtpServers()) do |used_server|
-        # if server is needed by pool.ntp.org and matches
-        if !Ops.get(needed_servers, used_server).nil?
-          Ops.set(needed_servers, used_server, true)
-        end
-      end
-
-      ret = true
-      Builtins.foreach(needed_servers) do |_nserver_name, ns_value|
-        ret = false if ns_value != true
-      end
-      ret
+      RANDOM_POOL_NTP_SERVERS.all? { |server| used_servers.include? server }
```

Comment:
np: According to our style guide, you should use a 1-letter identifier when using `{}` blocks.